File: TheActivityLifecyle/app/src/main/java/com/example/theactivitylifecyle/MainActivity.java
package com.example.theactivitylifecyle;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private static String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Log.i(TAG,"onCreate()");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        Log.i(TAG,"onRestart()");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Log.i(TAG,"onDestroy()");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG,"onStop()");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG,"onPause()");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG,"onResume()");
    }

//    @Override
//    protected void onPostResume() {
//        super.onPostResume();
//        Log.i(TAG,"onPostResume()");
//    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG,"onStart()");
    }
}

File: ShareViewModel/ColorGame/app/src/main/java/com/example/colorgame/MainActivity.kt
package com.example.colorgame

import android.content.Context
import android.os.Bundle
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.colorgame.fragments.ColorFragment
import com.example.colorgame.fragments.ColorSetFragment

class MainActivity : AppCompatActivity() {
    private lateinit var colorViewFragment: ColorFragment
    private lateinit var colorSetFragment: ColorSetFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setupColorReceiverFragment()
        setupColorSetFragment()
    }

    override fun onCreateView(
        parent: View?, name: String, context: Context,
        attrs: AttributeSet
    ): View? {
        return super.onCreateView(parent, name, context, attrs)
    }

    private fun setupColorReceiverFragment() {
        colorViewFragment = ColorFragment.newInstance()
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.receiverFragment, colorViewFragment)
        ft.commit()
    }

    private fun setupColorSetFragment() {
        colorSetFragment = ColorSetFragment.newInstance()
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.senderFragment, colorSetFragment)
        ft.commit()
    }
}